### ISSUE TITLE
pkglistgen: update_and_solve staging scope and drop list generator do not work for SLE-15

### ISSUE THREAD

**Issue by coolo** [openSUSE/osc-plugin-factory#1392]
```
~/osc-plugin-factory/pkglistgen.py -A https://api.suse.de update_and_solve -p SUSE:SLE-15:GA -s staging
[staging] SUSE:SLE-15:GA:Staging:A:DVD/standard: update and solve
Traceback (most recent call last):
  File "/home/leaper/osc-plugin-factory/pkglistgen.py", line 1329, in <module>
    sys.exit(app.main())
  File "/usr/lib/python2.7/site-packages/cmdln.py", line 257, in main
    return self.cmd(args)
  File "/usr/lib/python2.7/site-packages/cmdln.py", line 280, in cmd
    retval = self.onecmd(argv)
  File "/usr/lib/python2.7/site-packages/cmdln.py", line 412, in onecmd
    return self._dispatch_cmd(handler, argv)
  File "/usr/lib/python2.7/site-packages/cmdln.py", line 1100, in _dispatch_cmd
    return handler(argv[0], opts, *args)
  File "/home/leaper/osc-plugin-factory/pkglistgen.py", line 1072, in do_update_and_solve
    self.update_and_solve_target(apiurl, target_project, target_config, main_repo, opts_dvd, skip_release=True)
  File "/home/leaper/osc-plugin-factory/pkglistgen.py", line 1087, in update_and_solve_target
    packages = ET.parse(http_GET(url)).getroot()
  File "/usr/lib/python2.7/site-packages/osc/core.py", line 3355, in http_GET
    def http_GET(*args, **kwargs):    return http_request('GET', *args, **kwargs)
  File "/home/leaper/osc-plugin-factory/osclib/cache.py", line 38, in http_request
    ret = osc.core._http_request(method, url, headers, data, file)
  File "/usr/lib/python2.7/site-packages/osc/core.py", line 3344, in http_request
    fd = urlopen(req, data=data)
  File "/usr/lib64/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python2.7/urllib2.py", line 435, in open
    response = meth(req, response)
  File "/usr/lib64/python2.7/urllib2.py", line 548, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python2.7/urllib2.py", line 473, in error
    return self._call_chain(*args)
  File "/usr/lib64/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/usr/lib64/python2.7/urllib2.py", line 556, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 404: Not Found
```

**Comment by lnussel** [comment]
@coolo, does this help?

diff --git a/pkglistgen.py b/pkglistgen.py
index 9070f01..2c3bf73 100755
--- a/pkglistgen.py
+++ b/pkglistgen.py
@@ -1065,7 +1065,7 @@ class CommandLineInterface(ToolBase.CommandLineInterface):
                     self.options.repos.append('/'.join([opts.project, 'bootstrap_copy']))
 
                 # DVD project first since it depends on main.
-                if api.rings:
+                if self.item_exists(opts.project + ":DVD"):
                     opts_dvd = copy.deepcopy(opts)
                     opts_dvd.project += ':DVD'
                     self.options.repos.insert(0, '/'.join([opts_dvd.project, main_repo]))

**Comment by coolo** [comment]
s,self,api, and I at least run into the next error

**Comment by jberry-suse** [comment]
Hmm, I have the command in my history so clearly I ran it, but all staging must have been busy and I didn't notice or somesuch as it was merely a run it once to see if it runs. I published the first and will leave it running in background, but will take a while for `bs_mirrorfull`.

**Comment by jberry-suse** [comment]
Actually, it finished (wasn't too much since my last run) and seems to be happy. Presumably still some changes that need to be merged in that are SLE specific, but it runs #1329.
